File: app/src/main/java/com/example/myapplication/RetrofitClientInstance.java
package com.example.myapplication;

import com.google.gson.Gson;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.GET;

public class RetrofitClientInstance {

    private static final String BASE_URL = "https://api.flickr.com/services/rest/";

    private static Retrofit retrofit;
    private static OkHttpClient client = new OkHttpClient.Builder()
            .connectTimeout(100, TimeUnit.SECONDS)
            .readTimeout(100, TimeUnit.SECONDS).build();

    private static Retrofit getRetrofitInstance() {
        if (retrofit == null) {
            retrofit = new retrofit2.Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .client(client)
                    .addConverterFactory(GsonConverterFactory.create(new Gson()))
                    .build();
        }
        return retrofit;
    }

    public static Endpoints getService() {
        return getRetrofitInstance().create(Endpoints.class);
    }

    public interface Endpoints {

        @GET("?method=flickr.photos.search&api_key=3e7cc266ae2b0e0d78e279ce8e361736&format=json&nojsoncallback=1&text=tesla")
        Call<PhotoCollection> getAllPhotos();
    }
}

File: app/src/main/java/com/example/myapplication/RetrofitClientInstance.kt
package com.example.myapplication

import com.google.gson.Gson
import okhttp3.OkHttpClient
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import retrofit2.http.GET
import retrofit2.http.Query
import java.util.concurrent.TimeUnit

object RetrofitClientInstance {
    private const val BASE_URL = "https://api.flickr.com/services/rest/"
    private var retrofit: Retrofit? = null
    private val client = OkHttpClient.Builder()
            .connectTimeout(100, TimeUnit.SECONDS)
            .readTimeout(100, TimeUnit.SECONDS).build()

    private val retrofitInstance: Retrofit?
        get() {
            if (retrofit == null) {
                retrofit = Retrofit.Builder()
                        .baseUrl(BASE_URL)
                        .client(client)
                        .addConverterFactory(GsonConverterFactory.create(Gson()))
                        .build()
            }
            return retrofit
        }

    val service: Endpoints?
        get() = retrofitInstance?.create(Endpoints::class.java)

    interface Endpoints {
        @GET("?method=flickr.photos.search&api_key=3e7cc266ae2b0e0d78e279ce8e361736&format=json&nojsoncallback=1")
        fun getAllPhotos(@Query("text") text: String?): Call<PhotoCollection?>?
    }
}